jsonschema/exceptions.py

```diff
@@ -28,17 +32,17 @@ class _Error(Exception):
     def __init__(
         self,
         message: str,
-        validator=_unset,
-        path=(),
-        cause=None,
+        validator: str | _utils.Unset = _unset,
```

Comment:
I think these `_unsets` still need removing / `# type: ignore'ing` as per the above, I still don't want them appearing in docs (I don't want end-users knowing `_unset` exists at all even).

Reply:
Sorry for never getting back on this, life became busy.

I don't really know how to resolve this as I think it would start to become really hard to add typing to the package itself if we were to add `type: ignore` everywhere. If it is not part of the public API then it will be hard to "hide" it from users if it is also being used in parts that are exposed to the public API (such as the exception classes).

I'd be very happy to help add typing and refactor some code so that internal details don't get exposed, but adding incorrect typing and then adding `type: ignore` seems to more harm than good.

Reply:
Thanks for the update and no worries on the delay, sorry for my own.

To be direct, kind of like https://github.com/python-jsonschema/jsonschema/pull/1027#discussion_r1042505168 I'm not sure I'm ready to merge this as-is in that case -- I have to kind of stew on it a bit. I've hopefully said a few times before, I think for documentation and editor support that I'm quite happy to merge or add typing related fixes, but beyond the two I really honestly think it's all overall more annoyance than it helps still, so while I'm slightly sympathetic, I will always still go with the former (docs) over anything else -- I want to get some `$ref` related functional fixes out, which will likely be another few weeks, but then I can perhaps think a bit more about whether it's possible to get an improvement that I'm happier with. Just being transparent to not waste any of your time.